### PR TITLE
Push Artifacts to Azure Artifacts

### PR DESCRIPTION
**Pull request by ggallotti**
Issue:100507

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -78,31 +78,74 @@ jobs:
...
+            $PushToAzureArgs  += "--skip-duplicate"
+          }
+                    
+          dotnet $PushToAzureArgs
+          $pushedToAzure += 1
+          if (!$IsPrerelease) {
+            dotnet $PushToGitHubArgs
+            $pushedToGitHub += 1
+            dotnet $PushToNugetArgs
+            $pushedToNuget += 1
```
Comment (jechague) [inline .github/workflows/build.yml:119]: Should this package be pushed to NuGet Gallery and Azure Artifacts?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -78,31 +78,74 @@ jobs:
...
+            path: ./tmp/.github/actions/dispatch-build-result
+
+      - name: Dispatch build result
+        uses: ./tmp/.github/actions/dispatch-build-result
+        with:
+          component-name: ${{ github.event.inputs.repository }}
+          branch-ref: ${{ env.GIT_REF }}
+          new-version: ${{ needs.build.outputs.NUGET_VERSION }}
+          committer: ${{ needs.build.outputs.LAST_COMMITTER }}
+          commit-message: ${{ needs.build.outputs.COMMIT_MESSAGE }}
```
Comment (jechague) [inline .github/workflows/build.yml:155]: None of this outputs are defined in the workflow

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -78,31 +83,74 @@ jobs:
         committer: ${{ github.actor }}
         package-names: ${{ env.DispatchPackageNames }}
 
-  dispatch-workflow:
-    if: ${{ github.repository_owner == 'GeneXusLabs' && github.base_ref == '' && (github.ref == 'refs/heads/beta' || github.ref == 'refs/heads/master') }}
+    - name: Push packages
+      if: github.repository_owner == 'GeneXusLabs' && steps.buildVariables.outputs.SHOULD_DEPLOY == 'true'
```
Comment (jechague) [inline .github/workflows/build.yml:87]: The SHOULD_DEPLOY output is not defined

**Review by jechague — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (ggallotti) [reply .github/workflows/build.yml:111]: Is not Azure Artifacts the place we deploy all beta packages? Should I disable it?

Comment (jechague) [reply .github/workflows/build.yml:117]: Sorry, meant to say NuGet Gallery

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -2,20 +2,31 @@ name: Build
...
       Configuration: Release
 
     runs-on: [windows-latest]
 
+    outputs:
+      NUGET_VERSION: ${{ steps.buildVariables.outputs.NUGET_PACKAGE_VERSION }}
+      LAST_COMMITTER: ${{ steps.buildVariables.outputs.LAST_COMMITTER }}
+      COMMIT_MESSAGE: ${{ steps.buildVariables.outputs.COMMIT_MESSAGE }}
+      SHOULD_DEPLOY: ${{ steps.buildVariables.outputs.SHOULD_DEPLOY }}
+
```
Comment (jechague) [inline .github/workflows/build.yml:29]: A step with id "buildVariables" is missing where these 4 values should be calculated

File: .github/workflows/build.yml (hunk 1 above)
Comment (ggallotti) [reply .github/workflows/build.yml:117]: Nuget is not the place we deploy Release packages?

**Review by jechague — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -2,53 +2,87 @@ name: Build
...
 on:
   workflow_dispatch:
+    inputs:
+      skip-duplicates:
+          description: 'Whether to fail or skip duplicates when uploading to a package repository'
+          required: false
+          default: 'true'
   push:
-    branches: [ beta, master, release-* ]
+    branches: [ beta, master, release-*, push ]
```
Comment (jechague) [inline .github/workflows/build.yml:11]: What is the push branch?

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -78,31 +112,77 @@ jobs:
...
+        $IsPrerelease = [System.Convert]::ToBoolean($Env:IsPrerelease)
+        $totalPackages  = 0
+        $pushedToAzure  = 0
+        $pushedToGitHub = 0
+        $pushedToNuget  = 0                
+
+        Get-ChildItem ".\dotnet\*.nupkg" -Recurse | ForEach-Object {
+          $PushToGitHubArgs = @("nuget", "push", $_.FullName, "--source", $Env:GPRFeedURL, "--api-key", "${{ secrets.SECURE_TOKEN }}")
+          $PushToNugetArgs  = @("nuget", "push", $_.FullName, "--source", $Env:NuGetFeedURL, "--api-key", "${{ secrets.NUGET_ORG_TOKEN }}")
+          $PushToAzureArgs  = @("nuget", "push", $_.FullName, "--source", $Env:AZURE_ARTIFACTS_URL, "--api-key", "DUMMY-KEY")
```
Comment (jechague) [inline .github/workflows/build.yml:130]: AZURE_ARTIFACTS_URL env var is not defined

**Review by jechague — CHANGES_REQUESTED** [review]
Last two things to fix

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -16,93 +21,179 @@ jobs:
...
+          echo "::set-output name=LAST_COMMITTER::$LAST_COMMITTER"
+          echo "::set-output name=COMMIT_MESSAGE::$COMMIT_MESSAGE"
+
+      - name: Generate package
+        env:
+          CommitNumber: ${{env.COMMIT_NUMBER}}
+        run: dotnet pack $Env:Solution --configuration $Env:Configuration /p:Version=$Env:NUGET_PACKAGE_VERSION
+
+      - name: Checkout GeneXusLabs dispatch action
+        uses: actions/checkout@v3
```
Comment (jechague) [inline .github/workflows/build.yml:88]: The "Checkout GeneXusLabs dispatch action" step should not be in this job

**Review by jechague — CHANGES_REQUESTED** [review]

**Review by jechague — APPROVED** [review]

**Review by jechague — APPROVED** [review]

**Pull request merged**